### DISCUSSION TITLE
Splitting html output in several pages

### DISCUSSION THREAD

**Discussion by xtimbeau** [quarto-dev/quarto-cli#11132] [Feature Requests]
### Description

When building a website with technical content, with multiple .qmd document, some document may become long and stuffed with a large number of images, widgets and data. That can make the html file large enough to be problematic to charge.

One solution is to break down the .qmd into several pieces, at the cost of continuous numbering, inside crossrefs, common bibliography and so on. 

Another solution is to have a book with several chapters, bu at the cost of multiple separate documents as allowed by a website.

One alternate solution could be to have for a single, but long, .qmd the ability to render it as mulitple html with navigation button as for a book, keeping numbering and referencing as in a single document but spreaded over multiple pages. The split could be done on header level without changing the interface very much.

**Comment by mcanouil** [comment]
If you are concerned with the loading time of images inside a long HTML document, you probably want to enable image lazy loading.

> One solution is to break down the .qmd into several pieces, at the cost of continuous numbering, inside crossrefs, common bibliography and so on.

1. > [...] inside crossrefs

   You can build multiple Quarto documents and link them to each other (instead of cross-reference using `@`), that's what <https://quarto.org> does.

2. > [...] common bibliography

   Bibliography can be set at project, directory, or document level, so what's the issue/cost?

3. > [...] continuous numbering

   I don't understand. Your feature request is to split one Quarto document into several pages, you would not have "continuous numbering" anyway.


Would you mind sharing the context of this feature request / your use case? it would help see if there is something missing and the value it could bring.

**Comment by xtimbeau** [comment]
Lazy loading sounds interesting.

However, my motivation is not only loading time but also navigation and
presentation. We use quarto to publish quite long documents, and several of
them into a website structure. The website is well adapted to this, but the
long document is published as a long html page. I think it would be nice to
have the same page splitted with the book navigation.
An exemple of that is there : Prévision de l’OFCE de l’automne 2024 –
Prévision de l'OFCE, automne 2024 (sciences-po.fr)
<https://www.ofce.sciences-po.fr/prev/prev2409/>. You can access the 2 long
documents with the navbar (France, International). Most images aren't
images but html widgets produced by ggirafe, so I am not sure lazy loading
will change anything.

XT


Le lun. 21 oct. 2024 à 15:59, Mickaël Canouil ***@***.***> a
écrit :

**Reply by mcanouil** [reply]
Why not use [sidebar](https://quarto.org/docs/websites/website-navigation.html#side-navigation) in the long document with [bread-crumbs](https://quarto.org/docs/websites/website-navigation.html#breadcrumbs) and [page navigation](https://quarto.org/docs/websites/website-navigation.html#page-navigation)?
As done in the guides of <https://quarto.org>.

**Comment by xtimbeau** [comment]
Yes sure, I tested that and it works well. But to my understanding:
 a. there is no simple way to have the numbering of level 2 (or n) headings
as you have in a single .qmd (or a book)
 b. there is no way to keep the automatic and very convenient @fig cross
references with the preview when hovering on the link
 c. and it is difficult to produce a pdf with all documents gathered.
An automatic splitting would make easy the breadcrumb and page navigation
for html and keep the long continuous format for pdf.

Le lun. 21 oct. 2024 à 16:37, Mickaël Canouil ***@***.***> a
écrit :
